### ISSUE TITLE
Merge request

### ISSUE THREAD

**Issue by ienroll** [gmessner/gitlab4j-api#140]
Hello,
Is there a way to pull comments on the files that are  part of a given merge?

Thanks
Ravi

**Comment by gmessner** [comment]
@ienroll 
The CommitsApi has a method to get all the comments associated with a particular commit sha, is this what you need?

```java
/**
 * Get the comments of a commit in a project.
 *
 * GET /projects/:id/repository/commits/:sha/comments
 *
 * @param projectId the project ID that the commit belongs to
 * @param sha a commit hash or name of a branch or tag
 * @return a List of Comment instances for the specified project ID/sha pair
 * @throws GitLabApiException GitLabApiException if any exception occurs during execution
 */
public List<Comment> getComments(int projectId, String sha)

**Comment by ienroll** [comment]
Hello,
 I tried getComments API. It does not work for me.Here is how I tried:

gitLabApi.getCommitsApi().getComments(project.getId(), shaCd); -- returns empty list
gitLabApi.getCommitsApi().getComments(project.getId(), branchName); -- returns empty list

However, I could get comment of merge itself and the files that were part of the merge(using Diff):
Commit commit = gitLabApi.getCommitsApi().getCommit(project.getId(), sha)
gitLabApi.getCommitsApi().getDiff(project.getId(), commit.getId()); --> returns only the merge comment and the list of file Names that are part of this merge request.

I need a way to pull not the file names but the comments on those files that are part of merge.

Thanks
Ravi

**Comment by gmessner** [comment]
@ienroll 
I believe what you are looking for is this:
https://docs.gitlab.com/ee/api/notes.html#merge-requests

If it is I can add this to the NotesApi, right now the NotesApi is only doing issues.

**Comment by gmessner** [comment]
@ienroll 
I've added support and tests for merge request comments and released 4.8.3.  You'll find the methods in the NotesApi.  Make sure you are passing in merge request IID, not ID when making these calls.

**Comment by gmessner** [comment]
@ienroll 
I assume this addressed your problem, so am closing the issue.  If your issue was not resolved you can reopen this issue.
